### PR TITLE
Feature/HPOwTemplates

### PR DESCRIPTION
**Pull request by agrant23**

### CODE REVIEW

File: pages/coinbase/homepage.py [hunk 1]
```diff
@@ -1,45 +1,237 @@
...
 import config
 from pages.templates.elements import *
-
+from selenium.common.exceptions import NoSuchElementException
 
 
 class HomePage:
 
-    def __init__(self, driver):
+    def __init__(self, driver, dismissCookies=True):
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: This is an ambiguous parameter name; initially I thought you put an option in to remove cookies. Change it to something like close_cookie_banner.

Also only use camel case for class names.

File: pages/coinbase/homepage.py [hunk 2]
```diff
@@ -1,45 +1,237 @@
...
 
         driver.get(config.coinbase_domain)
-        self.driver = driver
+        cookie_dismiss_button = DialogueBox(driver,self.cookie_dismiss_button_loc,self.visible_dialogue_box_loc)
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: Don't add templates on the fly. If you need a new template make a suggestion and we'll make a ticket if it's necessary. You should be able to do what you need by using the button class. The click function waits for the element to be clickable, so there should be no issue making a dismiss button object and doing .click()

- [ ] change this to a button
- [ ] move the button declaration into `build_elements()` and move the `if dismissCookies:` after that.
- [x] rename to `cookie_banner_dismiss_button()` or, if it's the only dismiss button on the page, simply `dismiss_button()`

File: pages/templates/elements.py [hunk 3]
```diff
@@ -81,6 +81,30 @@ def input(self, keys): self.element().send_keys(keys)
     def clear(self): self.element().clear()
 
 
+class DialogueBox(Element):
```
Comment (MNadolsky) [inline pages/templates/elements.py]: Remove this.

If you want to save it keep a branch.

File: pages/coinbase/homepage.py [hunk 4]
```diff
@@ -1,45 +1,237 @@
...
-        self.prices_link =      Link(self.driver, self.prices_link_loc)
-        self.products_menu =    ReactiveMenu(self.driver,self.products_menu_loc)
-        self.company_menu =     ReactiveMenu(self.driver, self.company_menu_loc)
-        self.earn_crypto_link = Link(self.driver, self.earn_crypto_link_loc)
-        self.sign_in_link =     Link(self.driver, self.sign_in_link_loc)
-        self.header_get_started_button = Button(
-            self.driver, self.header_get_started_button_loc)
+    # A dismiss cookie dialogue box appears and hides elements, clicking
+    # the dismiss button performed in the init method fixes this issue  
+    visible_dialogue_box_loc =      (by.XPATH, "//div[@class='sc-fzoLsD gkNzOD']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: When you change cookie_dismiss_button to a button you shouldn't need this anymore. In this case remove this.

File: pages/coinbase/homepage.py [hunk 5]
```diff
@@ -1,45 +1,237 @@
...
+    cookie_dismiss_button_loc =     (by.XPATH, "//button[.='Dismiss']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: rename this locator along with the button as mentioned above.

File: pages/coinbase/homepage.py [hunk 6]
```diff
@@ -1,45 +1,237 @@
...
 
+    # NAVIGATION BAR
 
+    home_link_loc =                 (by.XPATH, "//a[@title='Home']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: I didn't mention this, but we're sticking to PEP 8 guidelines here and maintaining a 79 character/line limit. Please ensure no lines are longer than 79 characters. My editor is set to highlight all characters after the 79th in red; you can do something like that to help you if you like.

https://www.python.org/dev/peps/pep-0008/#indentation

File: pages/coinbase/homepage.py [hunk 7]
```diff
@@ -1,45 +1,237 @@
...
+    products_menu_loc =             (by.XPATH, "//button[contains(text(),'Products')]")
+    company_menu_loc =              (by.XPATH, "//button[contains(text(),'Company')]")
+    earn_crypto_link_loc =          (by.XPATH, "//a[@title='Earn crypto']")
+    sign_in_link_loc =              (by.XPATH, "//a[@title='Sign in']")
+    header_get_started_button_loc = (by.XPATH, "//a[.='Get started']")
+
+         
+    # CREATE ACCOUNT
+
+    get_started_email_field_loc =   (by.XPATH, "//input[@type='email']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: Name elements the following way:

1. append the element type to the end (e.g. 'field,' 'button,' etc.)
2. use the visible aspects of the element to name
3. if the visible aspects are not sufficient, prepend it with a descriptor based on visible aspects of the conceptual grouping of elements the element is in (e.g. `**header**_get_started_button()` to disambiguate from the other get started button)
4. if there are no appropriate visible aspects, use use an element attribute value like @id.

In this case, we would have gotten to step 2 and stopped: `email_address_field_loc()`. Please rename.

Additionally, please review your other names to ensure they obey these naming rules.

File: pages/coinbase/homepage.py [hunk 8]
```diff
@@ -1,45 +1,237 @@
...
+    get_started_button_loc =        (by.XPATH, "//input[@type='email']/following-sibling::button")
+
+    # CRYPTO CURRENCY MENU
+
+    bitcoin_text_link_loc =         (by.XPATH, "//h4[.='Bitcoin']")
+    bitcoin_BTC_link_loc =          (by.XPATH, "//h4[.='BTC']")
+    bitcoin_logo_link_loc =         (by.XPATH, "//img[@alt='Bitcoin logo']")
+    bitcoin_price_button_loc =      (by.XPATH, "//a[@href='/price/bitcoin']/button")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: For testing purposes I'm not a huge fan of finding buttons by @href. This can go wrong in too many ways depending on the direction of the wind and what brand of coffee the dev drank. Additionally, one of the things we might test is that the buttons go to the right place, and if someone mixed up the @href of two buttons your xcode will just end up finding the other button and clicking it, and going to the link you expected it to go to, and the test would pass even though it should fail.

THAT BEING SAID, you _should_ be able to find this button using the great grandparent element of this one, `//tr[contains(@class,'AssetTableRow')][1]`, but that's terrible xpath as well. In this case I would ask a dev to put an @id in here, but we can't do that. so..... 

Don't worry about it. Your xpath here is fine. Good job using what you had, it's probably what I would have done. I just wanted to mention that under normal circumstances (i.e., we can talk to the devs) we wouldn't do this.

File: pages/coinbase/homepage.py [hunk 9]
```diff
@@ -1,45 +1,237 @@
...
+    litecoin_LTC_link_loc =         (by.XPATH, "//h4[.='LTC']")
+    litecoin_logo_link_loc =        (by.XPATH, "//img[@alt='Litecoin logo']")
+    litecoin_price_button_loc =     (by.XPATH, "//a[@href='/price/litecoin']/button")
+
+    # EARNING LINKS BLCOK
+
+    start_earning_button_loc =       (by.XPATH, "//button[contains(@class,'EarnUpsellBanner')]")
+    start_earning_lower_button_loc = (by.XPATH, "//button[contains(@class,'EarnFooterBanner')]")
+    maker_link_loc =                 (by.XPATH, "//a[@href='/earn/maker']/div")
+    celo_link_loc =                  (by.XPATH, "//a[@href='/earn/celo']/div/div/h2")   #I don't love this xpath, but I feel it is the best way to locate this element
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: Why not just `//a[@to='/earn/celo']`? I mean that's also bad xpath for the reasons in my rant above, but given the circumstances I'll allow it. Why do you dig all the way down to the h2 element when you could just click the anchor?

File: pages/coinbase/homepage.py [hunk 10]
```diff
@@ -1,45 +1,237 @@
...
+    view_more_link_loc =             (by.XPATH, "//a[@href='/earn']/div")
+
+    # MOBIL APPS
+    
+    android_app_link_loc =           (by.XPATH, "//a[contains(@href,'android')]")#='Android']")
+    ios_app_link_loc =               (by.XPATH, "//a[.='iOS']")
+
+    # COINBASE SECURITY/INSURANCE
+
+    security_link_loc =             (by.XPATH,"//a[contains(@title,'funds') or contains(@title,'secure')]") #either funds or secure works' alone, I feel doing this increases longengevity
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: "funds" and "secure" could come up in sooooo many titles in the future. better to just `.=<the whole sentence>`

File: pages/coinbase/homepage.py [hunk 11]
```diff
@@ -1,45 +1,237 @@
...
+
+    start_earning_button_loc =       (by.XPATH, "//button[contains(@class,'EarnUpsellBanner')]")
+    start_earning_lower_button_loc = (by.XPATH, "//button[contains(@class,'EarnFooterBanner')]")
+    maker_link_loc =                 (by.XPATH, "//a[@href='/earn/maker']/div")
+    celo_link_loc =                  (by.XPATH, "//a[@href='/earn/celo']/div/div/h2")   #I don't love this xpath, but I feel it is the best way to locate this element
+    compound_link_loc =              (by.XPATH, "//a[@href='/earn/compound']/div/div/h2")   #ditto
+    EOS_link_loc =                   (by.XPATH, "//a[@href='/earn/eos']/div/div/h2")
+    view_more_link_loc =             (by.XPATH, "//a[@href='/earn']/div")
+
+    # MOBIL APPS
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: typo

File: pages/coinbase/homepage.py [hunk 12]
```diff
@@ -1,45 +1,237 @@
...
+    
+    android_app_link_loc =           (by.XPATH, "//a[contains(@href,'android')]")#='Android']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: Can you please show this to me later? can't find it.

File: pages/coinbase/homepage.py [hunk 13]
```diff
@@ -1,45 +1,237 @@
...
+    insurance_link_loc =            (by.XPATH,"//a[contains(@title,'insurance') or contains(@title,'policy')]")     #same as above
+    best_practices_link_loc =       (by.XPATH,"//a[contains(@title,'industry') or contains(@title,'practices')]")   #at first I thought I'd use an and instead of an or in case another title is created with just one of these words in it.
+
+    # LANGUAGE MENU FOOTER
+
+    language_menu_loc =             (by.XPATH,"//select[contains(@class,'Language')]")
+
+    # PRODUCTS FOOTER
+
+    coinbase_link_loc =             (by.XPATH, "//a[.='Coinbase']")     #from here down I feel like maybe this is an easy/lazy way to finish the homepage
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: What does this comment mean?

**Review by MNadolsky — CHANGES_REQUESTED** [review]
Hey I like your xpath.

File: pages/coinbase/homepage.py (hunk 2 above)
Comment (agrant23) [reply pages/coinbase/homepage.py]: I tried using the Button class and I had to create this DialogueBox element to perform this.

Comment (agrant23) [reply pages/coinbase/homepage.py:12]: OK!! I really did use Button when I did this and it didn't work..... But now it is working.... I don't know why!

Comment (agrant23) [reply pages/coinbase/homepage.py]: In the build elements method I commented out the same line that is in the init method for declaring the cookie_banner_dismiss_button. when I place a self on the cookie banner dimsiss button in the init method and use that line in the build elements method, I get a HomePage object has no attribute error for this button. Maybe I'm missing something, I know we can use attributes across methods by using the self in this manner but it's just not working for me in this case.  The way I have it works great. Please let me know if you find a way to do it that meets your criteria. I tried to use you Button class again and it still doesn't work. I am testing this by instantiating the HomePage class.

File: pages/coinbase/homepage.py (hunk 9 above)
Comment (agrant23) [reply pages/coinbase/homepage.py]: I think I forgot to go back and fix this xpath after I fixed the cookie banner getting in the way. So the anchor is what you call the highest parent level of an element?

File: pages/coinbase/homepage.py (hunk 12 above)
Comment (agrant23) [reply pages/coinbase/homepage.py]: image

image

File: pages/coinbase/homepage.py (hunk 13 above)
Comment (agrant23) [reply pages/coinbase/homepage.py]: I just felt that xpath was too easy, maybe I was being lazy. The rest of my xpaths are like this

File: pages/coinbase/homepage.py [hunk 14]
```diff
@@ -1,27 +1,37 @@
 
 import config
 from pages.templates.elements import *
-
+from selenium.common.exceptions import NoSuchElementException
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: You don't seem to be using this. remove.

File: pages/coinbase/homepage.py [hunk 15]
```diff
@@ -1,27 +1,37 @@
...
 
 
 class HomePage:
 
-    def __init__(self, driver, spawn=True):
+    def __init__(self, driver, spawn=True, close_cookie_banner=True):
         """
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py:11]: Document your new parameter.

File: pages/coinbase/homepage.py [hunk 16]
```diff
@@ -1,27 +1,37 @@
...
 
+        if close_cookie_banner:
+            wait(self.driver,10).until(EC.visibility_of_element_located(
+                self.visible_cookie_banner_loc))
+            self.cookie_banner_dismiss_button.click()
+
+
+    # A dismiss cookie dialogue box appears and hides elements, clicking
+    # the dismiss button performed in the init method fixes this issue  
+    visible_cookie_banner_loc = (by.XPATH, "//div[@class='sc-fzoLsD gkNzOD']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: This isn't the xpath for the banner, it's the xpath for the text on the banner. Rename this (to something like "banner_text_loc") or put in the xpath for the actual banner. I prefer the former, but if you run into problems don't worry about it too much.

Also don't use class names if they're random-seeming strings. These sometimes change randomly. Use the contained text instead ("Cookie Policy" should work fine, then you can work up to the banner element from there).

- [ ] remove the 'visible_' part of this (unless there's a good reason for it I'm not thinking of?)
- [x] rename the locator or use the xpath for the banner
- [x] don't use the class name to find the element

File: pages/coinbase/homepage.py [hunk 17]
```diff
@@ -30,27 +40,242 @@ def __init__(self, driver, spawn=True):
...
-    header_get_started_button_loc =(
-        by.XPATH, "//a[.='Get started']")
+    header_get_started_button_loc = (by.XPATH, "//a[.='Get started']")
+
+         
+    # CREATE ACCOUNT
+
+    email_address_field_loc = (by.XPATH, "//input[@type='email']")
+    get_started_button_loc =  (by.XPATH,
+                            "//input[@type='email']/following-sibling::button")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: Your indentation here doesn't match your indentation elsewhere.

File: pages/coinbase/homepage.py [hunk 18]
```diff
@@ -30,27 +40,242 @@ def __init__(self, driver, spawn=True):
...
+    view_more_link_loc =      (by.XPATH, "//a[@to='/earn']")
+
+    # MOBILE APPS
+    
+    android_app_link_loc =        (by.XPATH, "//a[contains(@href,'android')]")
+    ios_app_link_loc =            (by.XPATH, "//a[.='iOS']")
+
+    # COINBASE SECURITY/INSURANCE
+
+    secure_storage_link_loc =       (by.XPATH,
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: please fix the spacing

I'm wondering if it looks different in your IDE. If so you should probably do something about that.

File: pages/coinbase/homepage.py [hunk 19]
```diff
@@ -30,27 +40,242 @@ def __init__(self, driver, spawn=True):
...
+        "//a[@title='Learn how Coinbase keeps your funds safe and secure']")     
+    protected_insurance_link_loc =      (by.XPATH,
+        "//a[@title=" +
+        "'Learn how your crypto is covered by our insurance policy']")  #I see how to PEP 8 format for a long string but doing this breaks the XPATH, so these two lines went over the 79 characters
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: Isn't this fixed? If so, delete the comment; if not, fix and delete the comment.

File: pages/coinbase/homepage.py [hunk 20]
```diff
@@ -30,27 +40,242 @@ def __init__(self, driver, spawn=True):
...
+    ios_app_link_loc =            (by.XPATH, "//a[.='iOS']")
+
+    # COINBASE SECURITY/INSURANCE
+
+    secure_storage_link_loc =       (by.XPATH,
+        "//a[@title='Learn how Coinbase keeps your funds safe and secure']")     
+    protected_insurance_link_loc =      (by.XPATH,
+        "//a[@title=" +
+        "'Learn how your crypto is covered by our insurance policy']")  #I see how to PEP 8 format for a long string but doing this breaks the XPATH, so these two lines went over the 79 characters   
+    industry_best_practices_link_loc = (by.XPATH,"//a[@title="
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: You can break the xpath string into just two pieces instead of three, right? If so please do.

File: pages/coinbase/homepage.py [hunk 21]
```diff
@@ -30,27 +40,242 @@ def __init__(self, driver, spawn=True):
...
+    # SOCIAL FOOTER
+
+    blog_link_loc =                 (by.XPATH, "//a[.='Blog']")
+    twitter_link_loc =              (by.XPATH, "//a[.='Twitter']")
+    facebook_link_loc =             (by.XPATH, "//a[.='Facebook']")
+
 
     def build_elements(self):
 
+        # COOKIE BANNER
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: This section should be at the bottom because it's at the bottom of the page.

File: pages/coinbase/homepage.py [hunk 22]
```diff
@@ -30,27 +40,242 @@ def __init__(self, driver, spawn=True):
...
+                                         self.email_address_field_loc)
+        self.get_started_button =  Button(self.driver,
+                                          self.get_started_button_loc)
+
+        # CRYPTO CURRENCY MENU
+  
+        self.bitcoin_text_link =  Link(self.driver,self.bitcoin_text_link_loc)
+        self.bitcoin_BTC_link =   Link(self.driver,self.bitcoin_BTC_link_loc)
+        self.bitcoin_logo_link =  Link(self.driver,self.bitcoin_logo_link_loc)
+        self.bitcoin_buy_button = Button(
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py:182]: Your indentation is very inconsistent.

I recommend doing something that almost always works. I do this:

```
self.bitcoin_buy_button = Button(
    self.driver,self.bitcoin_buy_button_loc)
```

First line ends at open parenthesis, second line begins a single tab after the first. It nearly always fits, so I can do the same thing every time and it looks neater. You're the main author of this file so you don't have to do what I did, but you should pick something versatile and stick with it.

File: pages/coinbase/signin.py [hunk 23]
```diff
@@ -8,6 +8,15 @@
 
 class SigninPage:
 
+<<<<<<< HEAD
```
Comment (MNadolsky) [inline pages/coinbase/signin.py]: bro

File: pages/coinbase/homepage.py [hunk 24]
```diff
@@ -30,27 +40,249 @@ def __init__(self, driver, spawn=True):
...
+                                          self.start_earning_button_loc)
+        self.footer_start_earning_button = Button(self.driver,
+                                          self.footer_start_earning_button_loc)
+        self.maker_link =     Link(self.driver,self.maker_link_loc)
+        self.celo_link =      Link(self.driver,self.celo_link_loc)
+        self.compound_link =  Link(self.driver,self.compound_link_loc)
+        self.EOS_link =       Link(self.driver,self.EOS_link_loc)
+        self.view_more_link = Link(self.driver,self.view_more_link_loc)
+
+        # MOBIL APPS
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: typo

**Review by MNadolsky — CHANGES_REQUESTED** [review]
please address all comments

File: pages/coinbase/homepage.py (hunk 17 above)
Comment (agrant23) [reply pages/coinbase/homepage.py]: for line 50 this is indented as far as it can to fit in the 79 character range

File: pages/coinbase/homepage.py [hunk 25]
```diff
@@ -2,26 +2,39 @@
...
 
+        if close_cookie_banner:
+            wait(self.driver,10).until(EC.visibility_of_element_located(
+                self.visible_cookie_banner_dismiss_button_loc))
+            self.cookie_banner_dismiss_button.click()
+
+
+    # A dismiss cookie dialogue box appears and hides elements, clicking
+    # the dismiss button performed in the init method fixes this issue  
+    visible_cookie_banner_dismiss_button_loc = (by.XPATH,
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: remove `visible_` part from name

File: pages/coinbase/homepage.py [hunk 26]
```diff
@@ -30,27 +43,241 @@ def __init__(self, driver, spawn=True):
...
+    blog_link_loc =         (by.XPATH, "//a[.='Blog']")
+    twitter_link_loc =      (by.XPATH, "//a[.='Twitter']")
+    facebook_link_loc =     (by.XPATH, "//a[.='Facebook']")
+
 
     def build_elements(self):
 
         # NAVIGATION BAR
 
-        self.home_link = Link(
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py:40]: Why did you remove the destination parameters? please restore.

File: pages/coinbase/homepage.py [hunk 27]
```diff
@@ -2,26 +2,39 @@
...
         driver = selenium driver
         spawn: bool
-            The default use case is that the driver being passed in is new and 
+            The default use case is that the driver being passed in is new and
             needs to be navigated to the homepage. if spawn=False, it is assumed
             that the driver passed in is already at the homepage.
+        close_cookie_banner: bool
+            The default use case closes the cooker banner when the page is
+            opened. When close_cookie_banner=False the cookie banner reamins 
+            open and can hide elements of the page.
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: remove "and can hide elements of the page."

File: pages/coinbase/homepage.py [hunk 28]
```diff
@@ -30,27 +43,241 @@ def __init__(self, driver, spawn=True):
...
+                                     "//a[@href='/price/bitcoin-cash']/button")
+
+    litecoin_text_link_loc= (by.XPATH, "//h4[.='Litecoin']")
+    litecoin_LTC_link_loc = (by.XPATH, "//h4[.='LTC']")
+    litecoin_logo_link_loc= (by.XPATH, "//img[@alt='Litecoin logo']")
+    litecoin_buy_button_loc=(by.XPATH, "//a[@href='/price/litecoin']/button")
+
+    # EARNING LINKS BLCOK
+
+    start_earning_button_loc=(by.XPATH,
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: make your indentation more consistent throughout this file

File: pages/coinbase/homepage.py [hunk 29]
```diff
@@ -30,27 +43,241 @@ def __init__(self, driver, spawn=True):
...
+    view_more_link_loc =    (by.XPATH, "//a[@to='/earn']")
+
+    # MOBILE APPS
+    
+    android_app_link_loc =  (by.XPATH, "//a[contains(@href,'android')]")
+    ios_app_link_loc =      (by.XPATH, "//a[.='iOS']")
+
+    # COINBASE SECURITY/INSURANCE
+
+    secure_storage_link_loc=(by.XPATH,
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: put spaces around the =

**Review by MNadolsky — CHANGES_REQUESTED** [review]

File: pages/coinbase/homepage.py [hunk 30]
```diff
@@ -1,56 +1,278 @@
...
+    litecoin_logo_link_loc =  (by.XPATH, "//img[@alt='Litecoin logo']")
+    litecoin_buy_button_loc = (by.XPATH, "//a[@href='/price/litecoin']/button")
+
+    # EARNING LINKS BLCOK
+
+    start_earning_button_loc = (
+        by.XPATH, "//button[contains(@class,'EarnUpsellBanner')]")
+    footer_start_earning_button_loc = (
+        by.XPATH, "//button[contains(@class,'EarnFooterBanner')]")
+    alogrand_link_loc =               (by.XPATH,"//a[@href='/earn/algorand']")
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py:84]: did you forget these?

File: pages/coinbase/homepage.py [hunk 31]
```diff
@@ -1,56 +1,278 @@
...
-
         driver.get(config.coinbase_domain)
-        self.driver = driver
 
+        self.driver = driver
         self.build_elements()
 
+        if close_cookie_banner:
+            driver.find_element(
+            *self.cookie_banner_dismiss_button_loc).send_keys(Keys.ENTER)   #The send keys is needed to 'dig out' the button from the cookie banner
```
Comment (MNadolsky) [inline pages/coinbase/homepage.py]: too many characters; put the comment above this line

**Review by MNadolsky — APPROVED** [review]
